aws/credentials/process_provider.go

```diff
@@ -0,0 +1,228 @@
+	}
+	return p.credentialProcess(process.String())
+}
+
+type credentialProcessResponse struct {
+	Version         int
+	AccessKeyID     string `json:"AccessKeyId"`
+	SecretAccessKey string
+	SessionToken    string
+	Expiration      string
+}
+
+func executeCredentialProcess(process string) ([]byte, error) {
+	processArgs := strings.Split(process, " ")
+	var command string
+	var cmdArgs []string
+
+	if len(processArgs) == 1 {
+		command = processArgs[0]
+	} else if len(processArgs) > 1 {
+		command = processArgs[0]
+		cmdArgs = processArgs[1:]
+	}
+
+	// TODO: Check for executability?
+	_, err := os.Stat(command)
+	if os.IsNotExist(err) {
+		return nil, awserr.New("ProcessProviderLoad", "credential_process "+command+" not found", err)
+	} else if err != nil {
+		return nil, awserr.New("ProcessProviderLoad", "failed to open credential_process "+command, err)
+	}
+
+	// Execute command
+	cmd := exec.Command(command, cmdArgs...)
+	cmd.Env = os.Environ()
+	out, err := cmd.Output()
+	if err != nil {
+		return nil, awserr.New("ProcessProviderLoad", "Error executing credential_process", err)
+	}
+	return out, nil
+}
+
+func (p *ProcessProvider) credentialProcess(process string) (Value, error) {
+	out, err := p.executionFunc(process)
+	if err != nil {
+		return Value{ProviderName: ProcessProviderName}, err
+	}
+
+	// Serialize and validate response
+	resp := &credentialProcessResponse{}
+	err = json.Unmarshal(out, resp)
+	if err != nil {
+		return Value{ProviderName: ProcessProviderName}, awserr.New("ProcessProviderLoad", "Error parsing credential_process output: "+string(out), err)
+	}
+	if resp.Version != 1 {
+		return Value{ProviderName: ProcessProviderName}, awserr.New("ProcessProviderLoad", " Version in credential_process output is not 1", err)
+	}
+	if len(resp.AccessKeyID) == 0 {
+		return Value{ProviderName: ProcessProviderName}, awserr.New("ProcessProviderLoad", " Missing required AccessKeyId in credential_process output", err)
+	}
+	if len(resp.SecretAccessKey) == 0 {
+		return Value{ProviderName: ProcessProviderName}, awserr.New("ProcessProviderLoad", " Missing required SecretAccessKey in credential_process output", err)
+	}
+
+	// Handle expiration
+	if len(resp.Expiration) > 0 {
+		expiry, err := time.Parse(time.RFC3339, resp.Expiration)
+		if err != nil {
+			return Value{ProviderName: ProcessProviderName}, awserr.New("ProcessProviderLoad", "Error parsing expiration of credential_process output: "+resp.Expiration, err)
+		}
+		p.SetExpiration(expiry, p.ExpiryWindow)
+	}
+
+	return Value{
+		ProviderName:    ProcessProviderName,
+		AccessKeyID:     resp.AccessKeyID,
+		SecretAccessKey: resp.SecretAccessKey,
+		SessionToken:    resp.SessionToken,
+	}, nil
+
+}
+
+// filename returns the filename to use to read AWS shared credentials.
+//
+// Will return an error if the user's home directory path cannot be found.
+func (p *ProcessProvider) filename() (string, error) {
```

Comment:
This information should come from the SDK's `session` similar to `stscreds`  There are several locations, and various env var names that can drive the config file to load within the SDK. Having loading the shared config file utility do this work, will help reduce duplication.